### DISCUSSION TITLE
Are CamPemPath/ServerPemPath parameters required for MilvusClient if the Milvus cert has been imported into the java trust store?

### DISCUSSION THREAD

**Discussion by ndtreviv** [milvus-io/milvus#34160] [Q&A and General discussion] [answered]
If I want to communicate with a Milvus over TLS, can I import the pem into the default java trust store or do I have to configure the client with the CaPemPath/ServerPemPath?

IE: at the moment I'm doing this:

```
        ConnectParam.Builder builder = ConnectParam
                .newBuilder()
                .withHost(host)
                .withPort(port)
                .withSecure(secure);

        if (StringUtils.isNotBlank(caPemPath)) {
            LOGGER.info("Using ca pem path: {}", caPemPath);
            builder.withCaPemPath(caPemPath);
        }
        if (StringUtils.isNoneBlank(serverPemPath, serverName)) {
            LOGGER.info("Using server pem path: {} and server name: {}", serverPemPath, serverName);
            builder.withServerPemPath(serverPemPath).withServerName(serverName);
        } else if (!StringUtils.isAllBlank(serverPemPath, serverName)) {
            throw new IllegalArgumentException("Server milvus.server-pem-path and milvus.server-name must both be set or both be blank");
        }
```

But if I've imported Milvus' pem into the java trust store, do I need to do any of this at all? Or will it work without it?

**Comment by yhmo** [comment, marked as answer]
The logic to build connections in Java SDK is here: https://github.com/milvus-io/milvus-sdk-java/blob/c8c592402123632c817e4d3c336c750d156da8fb/src/main/java/io/milvus/client/MilvusServiceClient.java#L102

I think you have to configure the client with the CaPemPath/ServerPemPath.
I am not sure what should be done to support Java trust store. If you know how to implement it, you are welcome to submit PR to the Java SDK repo.

**Reply by ndtreviv** [reply]
I see. Thanks for this. I can see it's possible, but because it's grpc and Netty then we have to do something special (like this: https://stackoverflow.com/a/64017014/498391)

I'll have a think about a PR!
